addon/models/resource.js

```diff
@@ -415,8 +418,14 @@ Resource.reopenClass({
     for (let i = 0; i < attrs.length; i++) {
       prototype[attrs[i]] = {};
     }
+
     const instance = this._super(prototype);
     if (properties) {
+      if (properties.hasOwnProperty('id')) {
+        // JSONAPI ids MUST be strings.
+        properties.id = properties.id.toString();
```

Comment:
This should work for both creating on client side (via app, e.g. new resource) as well as when deserializing, nice.